quinn-proto/src/frame.rs

```diff
@@ -841,9 +842,31 @@ pub const RETIRE_CONNECTION_ID_SIZE_BOUND: usize = 9;
 
 /// An unreliable datagram
 #[derive(Debug, Clone)]
-pub struct Datagram {
-    /// Payload
-    pub data: Bytes,
+pub enum Datagram {
+    /// Used to designate an outgoing immutable buffer
+    Outgoing(Bytes),
+    /// Used to designate an incoming mutable buffer
+    Incoming(BytesMut),
+}
+
+impl Datagram {
+    pub(crate) fn assert_incoming(self) -> BytesMut {
```

Comment:
Let's not panic here. Instead, call the function `incoming()` and have it return `Option<BytesMut>`.

Reply:
Okay